gh-103987: fix crash in mmap module (#103990)
python/cpython: Misc/NEWS.d/next/Library/2023-04-29-18-23-16.gh-issue-103987.sRgALL.rst

> Fix ``mmap_subscript`` in mmapmodule.c to avoid segmentation fault

Reviewer: ```suggestion
add more `CHECK_VALID`s in `mmapmodule.c` to avoid the file being invalidated by Python code.
```

> add more `CHECK_VALID`s in `mmapmodule.c` to avoid the file being invalidated by Python code.

Reviewer: ```suggestion
add more ``CHECK_VALID``s in ``mmapmodule.c`` to avoid the file being invalidated by Python code.
```

> add more ``CHECK_VALID``s in ``mmapmodule.c`` to avoid the file being invalidated by Python code.

Reviewer: ```suggestion
add more ``CHECK_VALID`` s in ``mmapmodule.c`` to avoid the file being invalidated by Python code.
```

> add more ``CHECK_VALID`` s in ``mmapmodule.c`` to avoid the file being invalidated by Python code.

Reviewer 2: Focus on what this means to users, not on the mechanism used to achieve it.
```suggestion
In :mod:`mmap`, fix several bugs that could lead to access to memory-mapped files after
they have been invalidated.
```
